### PR TITLE
feat(ai): Migrate ai features to use new consolidated flag

### PR DESCRIPTION
**Pull request by jennmueng**
Consolidate all the AI features to use the same flag. Also removes unused instances of the codebase indexing endpoints.

To ensure consistency for customers flagged by this, this must be merged after https://github.com/getsentry/sentry-options-automator/pull/2700

### CODE REVIEW

File: src/sentry/api/endpoints/group_ai_autofix.py [hunk 1]
```diff
@@ -161,9 +156,8 @@ def post(self, request: Request, group: Group) -> Response:
         created_at = datetime.now().isoformat()
 
         if not (
-            features.has("projects:ai-autofix", group.project)
-            or features.has("organizations:autofix", group.organization)
-            or group.organization.get_option("sentry:gen_ai_consent_v2024_11_14", False)
```
Comment (jennmueng) [inline src/sentry/api/endpoints/group_ai_autofix.py:166]: this should've been an `and` for the genai consent this whole time

Comment (roaga) [reply src/sentry/api/endpoints/group_ai_autofix.py:166]: No, we made this an `or` for the Copilot extension as a workaround to allow GitHub to give access to whoever without us having to flag them in.

File: src/sentry/api/endpoints/group_ai_autofix.py [hunk 2]
```diff
@@ -111,11 +111,6 @@ def _call_autofix(
                     else None
                 ),
                 "options": {
-                    "disable_codebase_indexing": features.has(
-                        "organizations:autofix-disable-codebase-indexing",
```
Comment (roaga) [inline src/sentry/api/endpoints/group_ai_autofix.py:115]: are we removing these flags elsewhere in another PR?

Comment (jennmueng) [reply src/sentry/api/endpoints/group_ai_autofix.py:115]: yes following up with a flag removal pr

File: src/sentry/api/endpoints/group_ai_autofix.py (hunk 1 above)
Comment (jennmueng) [reply src/sentry/api/endpoints/group_ai_autofix.py:166]: I see, but now is it ok as we're open beta-ing this or should we revert it to an or?

Comment (roaga) [reply src/sentry/api/endpoints/group_ai_autofix.py:166]: Offline decision: it's okay to remove this

**Review by roaga — APPROVED** [review]
Do we need to improve test coverage with all these changes?

**Comment by jennmueng** [comment]
@roaga yeah means the lines we change weren't covered

**Pull request merged**